torch/_inductor/fx_passes/misc_patterns.py

```diff
@@ -69,6 +69,52 @@ def randperm_index_replacement(x, slice_shape):
         scalar_workaround={"slice_shape": 42},
     )
 
+    # Float8 training patterns
+    E4M3_MAX_POS = torch.finfo(torch.float8_e4m3fn).max
+    E5M2_MAX_POS = torch.finfo(torch.float8_e5m2).max
+
+    def amax_with_scaling_pattern(tensor_x_inp, scale_x, IS_BWD):
+        tensor_x = tensor_x_inp.to(torch.float32) * scale_x
+        if IS_BWD:
+            tensor_x = tensor_x.clamp(min=-1 * E5M2_MAX_POS, max=E5M2_MAX_POS)
+            tensor_x = tensor_x.to(torch.float8_e5m2)
+        else:
+            tensor_x = tensor_x.clamp(min=-1 * E4M3_MAX_POS, max=E4M3_MAX_POS)
+            tensor_x = tensor_x.to(torch.float8_e4m3fn)
+        amax = torch.max(torch.abs(tensor_x_inp))
+        return (tensor_x, amax)
+
+    def amax_with_scaling_tiled_replacement(tensor_x_inp, scale_x, IS_BWD):
+        tensor_x = tensor_x_inp.to(torch.float32) * scale_x
+        if IS_BWD:
+            tensor_x = tensor_x.clamp(min=-1 * E5M2_MAX_POS, max=E5M2_MAX_POS)
+            tensor_x = tensor_x.to(torch.float8_e5m2)
+        else:
+            tensor_x = tensor_x.clamp(min=-1 * E4M3_MAX_POS, max=E4M3_MAX_POS)
+            tensor_x = tensor_x.to(torch.float8_e4m3fn)
+        amax_1 = torch.max(torch.abs(tensor_x_inp), dim=-1).values
+        amax = torch.max(amax_1)
+        return (tensor_x, amax)
+    
+    # The amax_with_scaling_pattern will also match dynamic scaling cases, we want to avoid that.
```

Comment:
is this expected?  The pattern for delayed scaling should be different from that of dynamic scaling:

```
# delayed pseudocode
x1, s1 = cast_and_get_new_scale(x0, s0)

# dynamic pseudocode
s1 = get_scale(x0)
x1 = cast(x0, s1)
```

Reply:
hmm, yea, it's surprising to me this is occurring. is there any other way to differentiate ? maybe based on uses of the scale, or amax ?